vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringBootAutoConfigurationConditionalTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.spring;
+
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.context.WebApplicationContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SpringBootAutoConfigurationConditionalTest {
+
+    @Test
+    public void customServletRegistrationBean() {
+        new WebApplicationContextRunner()
+                .withConfiguration(AutoConfigurations
+                        .of(SpringBootAutoConfiguration.class))
+                .withUserConfiguration(
+                        ServletRegistrationBeanConfiguration.class)
+                .run(context -> assertThat(context)
+                        .getBean(ServletRegistrationBean.class).isInstanceOf(
+                                ServletRegistrationBeanConfiguration.MockServletRegistrationBean.class));
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    public static class ServletRegistrationBeanConfiguration {
+
+        @Bean
+        @Autowired
```

Comment:
nit: IIRC Autowired annotation should not be required anymore
```suggestion
        @Autowired
```

Reply:
Removed